server/webapp/WEB-INF/rails.new/config/routes.rb

```diff
@@ -228,6 +228,9 @@
         patch :update, on: :member
       end
 
+      get 'stages/:pipeline_name/:pipeline_counter/:stage_name/:stage_counter/instance' => 'stages#show', constraints: {pipeline_name: PIPELINE_NAME_FORMAT, pipeline_counter: PIPELINE_COUNTER_FORMAT, stage_name: STAGE_NAME_FORMAT, stage_counter: STAGE_COUNTER_FORMAT}, as: :stage_instance_by_counter_api
+      get 'stages/:pipeline_name/:stage_name/history' => 'stages#history', constraints: {pipeline_name: PIPELINE_NAME_FORMAT, stage_name: STAGE_NAME_FORMAT}, as: :stage_history_api
+
```

Comment:
> @arvindsv - do we really need these constraints?

I guess so. I mean, it's not wrong. Also, it does make sense to get rid of invalid names at this level, than getting to the config or db and throwing them out, I suppose. What am I missing?